plugins/org.yakindu.sct.model.stext/src/org/yakindu/sct/model/stext/validation/STextJavaValidator.java

```diff
@@ -453,11 +453,13 @@ public void checkValueOfNoEvent(EventValueReferenceExpression exp) {
 		}
 
 		if (element != null && (!(element instanceof Event))) {
-			String elementName = "";
+			String elementName;
 			if (element instanceof NamedElement) {
 				elementName = "'" + ((NamedElement) element).getName() + "' ";
+			} else {
+				elementName = "Could not find event declaration.";
 			}
-			error(elementName + "is no event.", StextPackage.Literals.EVENT_VALUE_REFERENCE_EXPRESSION__VALUE, 0,
+			error(elementName, StextPackage.Literals.EVENT_VALUE_REFERENCE_EXPRESSION__VALUE, 0,
```

Comment:
In case of NamedElement, error message is only elementName now. Add "is no event" for example.